### PR TITLE
feat: add message endpoint, implement e2e test

### PR DESCRIPTION
**Pull request by dagmawig**
working on issue #2 

Add `DATABASE_URL` to .env file and run `test-e2e` script to do the message test.

The test sends a job to a queue with a `job name` and `payload` and fetches the job afterwards to confirm the job created.
The job is found in `pgboss` schema under `job` table.

image

image

### CODE REVIEW

File: __tests__/e2e/message.spec.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+            const response = await supertest(app).post('/message').send({jobName: 'add-tree-test', payload: {treeId: '0101010101'}});
+            expect(response.status).toBe(200); // eslint-disable-line
+            expect(response.body.job.name).toEqual('add-tree-test'); // eslint-disable-line
+            expect(response.body.job.data).toMatchObject({ // eslint-disable-line
+                treeId: '0101010101'
+            }) 
+        },
+        1000 * 30
+    )
+})
```
Comment (dadiorchen) [inline __tests__/e2e/message.spec.js]: @dagmawig please test the receiving, that is to say, you should subscribe the message queue name: 'add-tree-test' and make sure the message that you sent is received by the client.

like this:

```
it('test', done => {
  const client = new pgboss.client();
  client.subscribe('add-tree-test', data => {
    expect(data).toMatchObject({message:'ssss'});
    done();
   });
   // do the post

}}
```

**Review by dadiorchen — COMMENTED** [review]
This is awesome! The pgboss did exactly what I wanted: use jsonb to store the message, so in this way, in the future, we can query the message by filter easily, this is important to us.

**Comment by dagmawig** [comment]
In this fix instead of just sending job to queue, it also completes the job and confirms the job completed.

image

**Comment by dagmawig** [comment]
@dadiorchen  I have replaced pg-boss with postgres listen-notify client system. For the e-2-e test I have created two separate pg clients and confirmed they both receive the notification of record insertion. We can add more functionalities in separate PRs later but I believe this will take care of the need for a proof of concept. Please review and let me know what you think.

File: __tests__/e2e/message.spec.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        //  tree object to be added to table
+        const treeObj = {
+            latest_capture_id: uuid.v4(),
+            age: 1
+        }
+
+        // query to create notification function on postgres schema
+        let funQueryRes;
+        try {
+            funQueryRes = await pgClient.query(`CREATE OR REPLACE FUNCTION queue.new_tree_notify_trigger() RETURNS TRIGGER AS $$
```
Comment (dadiorchen) [inline __tests__/e2e/message.spec.js]: @dagmawig you are using `trigger` and I don't think this is good for our pub/sub model, it bind the message to the business model( trees here), for a pub/sub, it is business model agnostic and could be sent message by any client (node.js microservice), but trigger just work for tables and any db operation.

What I imagine is more like this topic: https://medium.com/@simon.white/postgres-publish-subscribe-with-nodejs-996a7e45f88

The rough test steps:
1. Create notify channel
2. To client to subscribe to the channel
3. Send message to channel by this test.
4. All the two clients received the message.

**Review by dadiorchen — CHANGES_REQUESTED** [review]

**Comment by dagmawig** [comment]
Below is what the `message` table looks like. Based on the value under the `name` column, the `queue` schema function sends the message to the appropriate channel. 
image

**Review by dadiorchen — APPROVED** [review]

**Pull request merged**